api/envoy/config/filter/http/grpc_shim/v2alpha1/grpc_shim.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package envoy.extensions.filter.http.grpc_shim.v2alpha1;
+option go_package = "v2";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Extensions gRPC Shim]
+// gRPC shim filter configuration
+message GrpcShim {
+  // The content-type to pass to the upstream when the gRPC shim filter is applied.
+  // The filter will also validate that the upstream responds with the same content type.
+  string content_type = 1 [(validate.rules).string.min_bytes = 1];
+
+  // If true, Envoy will assume that the upstream doesn't understand gRPC frames and
+  // strip the gRPC frame from the request, and add it back in to the response. This will
+  // hide the gRPC semantics from the upstream, allowing it to receive and respond with a
+  // simple binary encoded protobuf.
+  bool withhold_grpc_frames = 2;
```

Comment:
I like having this defaulted to false because it means that both the http1 bridge and this leaves the frame header alone by default.